lib/phoenix_live_view/router.ex

```diff
@@ -104,13 +115,123 @@ defmodule Phoenix.LiveView.Router do
   """
   defmacro live(path, live_view, action \\ nil, opts \\ []) do
     quote bind_quoted: binding() do
+      vsn = Phoenix.LiveView.Router.__session_vsn__()
+      default = {:default, %{session: %{}}, vsn}
+      live_session = Module.get_attribute(__MODULE__, :phoenix_live_session_current, default)
+
       {action, router_options} =
-        Phoenix.LiveView.Router.__live__(__MODULE__, live_view, action, opts)
+        Phoenix.LiveView.Router.__live__(__MODULE__, live_view, action, live_session, opts)
 
       Phoenix.Router.get(path, Phoenix.LiveView.Plug, action, router_options)
     end
   end
 
+  @doc """
+  TODO
+  """
+  defmacro live_session(name, do: block) do
+    quote do
+      live_session(unquote(name), [], do: unquote(block))
+    end
+  end
+
+  defmacro live_session(name, opts, do: block) do
+    quote do
+      Module.register_attribute(__MODULE__, :phoenix_live_sessions, accumulate: true)
+
+      {name, extra, vsn} =
+        unquote(__MODULE__).__live_session__(
+          __MODULE__,
+          unquote(opts),
+          unquote(name),
+          unquote(__MODULE__).__session_vsn__()
+        )
+
+      @phoenix_live_session_current {name, extra, vsn}
+      @phoenix_live_sessions {name, extra, vsn}
```

Comment:
You can also move the setting of these attributes and the register attribute call to inside `__live_session__`.